pkgs/build-support/dotnet/build-dotnet-module/fetch-deps.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, buildDotnetModule
+, dotnetValidateLockfileHook
+, nuget-to-nix
+
+, nugetSha256
+, src
+, pname
+, dotnet-sdk
+, projectFile
+, testProjectFile
+, dotnetFlags
+, dotnetRestoreFlags
+, enableParallelBuilding
+}:
+
+buildDotnetModule rec {
+  name = "${pname}-nuget-lockfile";
+
+  inherit src dotnet-sdk projectFile testProjectFile dotnetFlags dotnetRestoreFlags enableParallelBuilding;
+
+  nativeBuildInputs = [ dotnetValidateLockfileHook ];
+
+  generateLockfile = true;
+  dontSetNugetSource = true;
+  dontDotnetFixup = true;
+
+  impureEnvVars = lib.fetchers.proxyImpureEnvVars;
+  outputHashAlgo = "sha256";
+  outputHash = if (nugetSha256 != null)
+    then nugetSha256
+    else ""; # This needs to be set for networking, an empty string prints the "empty hash found" warning
+
+  preConfigure = ''
+    export PACKAGE_DIR=$(mktemp -d)
+
+    dotnetRestoreFlags+=(
+      --packages "$PACKAGE_DIR/nuget_pkgs"
+    )
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    deps_file="$PACKAGE_DIR/deps.nix"
+
+    echo "Writing lockfile..."
+    ${nuget-to-nix}/bin/nuget-to-nix "$PACKAGE_DIR/nuget_pkgs" > "$deps_file"
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    cp $deps_file $out
+    echo "Installed lockfile to: $out"
+
+    runHook postInstall
+  '';
+
+  postFixup = lib.optionalString (nugetSha256 == null) ''
+    echo "Please set nugetSha256 to the hash below!"
+  '';
+
+  meta.description = "A lockfile containing the Nuget dependencies for ${pname}";
```

Comment:
Should we inherit meta from the derivation?

Reply:
Sounds like a good idea, I think it makes sense to inherit `platforms` and `maintainers` but I'm not sure about other attributes. What do you think?

Reply:
I would just inherit everything and overwrite description but whatever you like more.